core/src/main/java/com/amazon/opendistroforelasticsearch/sql/analysis/Analyzer.java

```diff
@@ -118,6 +119,17 @@ public LogicalPlan visitRelation(Relation node, AnalysisContext context) {
     return new LogicalRelation(node.getTableName());
   }
 
+  @Override
+  public LogicalPlan visitRelationSubquery(RelationSubquery node, AnalysisContext context) {
+    context.push();
```

Comment:
Since we can support subquery now, do we need to pop up context somewhere?

Reply:
Just took another look, so context is pushed first when visit RelationSubquery, should we do analyzer recursion before this? In this way, symbol defined by outer won't be accessible to inner, ex. select ... from (select ... from B) as A, is A accessible when visit subquery B?